geomstats/geometry/sasaki_metric.py

```diff
@@ -0,0 +1,279 @@
+"""Class for the Sasaki metric.
+
+A class implementing the Sasaki metric: The natural metric on the tangent bundle TM
+of a Riemannian manifold M.
+
+Lead authors: E. Nava-Yazdani, F. Ambellan, M. Hanik and C. von Tycowicz.
+"""
+import os
+
+from joblib import Parallel, delayed
+
+import geomstats.backend as gs
+from geomstats.geometry.riemannian_metric import RiemannianMetric
+
+N_STEPS = 3
+
+
+def _gradient_descent(x_ini, i_pt, e_pt, grad, exp, lrate=0.1, max_iter=100, tol=1e-6):
+    """Apply a gradient descent until max_iter or a given tolerance is reached."""
+    x = x_ini
+    for _ in range(max_iter):
+        grad_x = grad(x, i_pt, e_pt)
+        grad_norm = gs.linalg.norm(grad_x)
+        if grad_norm < tol:
+            break
+        grad_x = -lrate * grad_x
+        x = exp(grad_x, x)
+    return x
+
+
+class SasakiMetric(RiemannianMetric):
+    """Implements of the Sasaki metric on the tangent bundle TM of a Riem. manifold M.
+
+    The Sasaki metric is characterized by the following three properties:
+     * the canonical projection of TM becomes a Riemannian submersion,
+     * parallel vector fields along curves are orthogonal to their fibres, and
+     * its restriction to any tangent space is Euclidean.
+
+    Geodesic computations are realized via a discrete formulation of the geodesic
+    equation on TM that involve geodesics, parallel translation, and the curvature
+    tensor on the base manifold M (see [1] for details).
+    However, as the implemented energy in the discrete-geodesics-optimization as well as
+    the approximations of its gradient slightly differ from those proposed in [1], we
+    also refer to [2] for additional details.
+
+    Parameters
+    ----------
+    metric : RiemannianMetric
+        Metric of the base manifold of the tangent bundle.
+
+    References
+    ----------
+    .. [1] Muralidharan, P., & Fletcher, P. T. (2012, June).
+    Sasaki metrics for analysis of longitudinal data on manifolds.
+    In 2012 IEEE conference on computer vision and pattern recognition (pp. 1027-1034).
+    https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4270017/
+    [2] Nava-Yazdani, E., Hanik, M., Ambellan, F., & von Tycowicz, C. (2022, June).
+    On Gradient Formulas in an Algorithm for the Logarithm of the Sasaki Metric
+    Technical Report Zuse-Institut Berlin (no. ZIB-22-12) (urn:nbn:de:0297-zib-87174).
+    https://nbn-resolving.org/urn/resolver.pl?urn:nbn:de:0297-zib-87174
+    """
+
+    def __init__(self, metric: RiemannianMetric):
+        self.metric = metric
+        shape = (2, gs.prod(gs.array(metric.shape)))
+
+        self.n_jobs = (
```

Comment:
Pass `n_jobs` as input with default value 1 (for consistency with other uses of `joblib` in the library).